diff --git a/src/Makefile_MT b/src/Makefile_MT
index a5c72c6..5e05c8e 100755
--- a/src/Makefile_MT
+++ b/src/Makefile_MT
@@ -6,7 +6,7 @@
 # in terminal to run PARDISO with 2 cpus
 # More info in ccx user manual section 2
 
-LIB_PATH=../../..
+LIB_PATH=../lib
 
 WITH_EXODUSII = false
 WITH_MKL = true
@@ -24,7 +24,7 @@ CFLAGS+= -I $(LIB_PATH)/ARPACK -DARPACK
 
 #SPOOLES
 #CFLAGS+= -I /home/dalab/CalculixBuild/SPOOLES.2.2 -DSPOOLES
-CFLAGS+= -I $(LIB_PATH)/SPOOLES.2.2 -DSPOOLES
+CFLAGS+= -I $(LIB_PATH)/SPOOLES -DSPOOLES
 
 #BLAS and LAPACK
 LDFLAGS= -llapack -lblas
@@ -84,8 +84,8 @@ OCCXMAIN = $(SCCXMAIN:.c=.o)
 
 LIBS = \
        $(LIB_PATH)/ARPACK/libarpack_INTEL.a \
-       $(LIB_PATH)/SPOOLES.2.2/MT/src/spoolesMT.a \
-       $(LIB_PATH)/SPOOLES.2.2/spooles.a
+       $(LIB_PATH)/SPOOLES/MT/src/spoolesMT.a \
+       $(LIB_PATH)/SPOOLES/spooles.a
 
 all: ccx_2.19_MT
 
